### DISCUSSION TITLE
OIDC with ADFS Email not included in Profile

### DISCUSSION THREAD

**Discussion by PowellTravis** [outline/outline#5756] [Self Hosting]
I have been working to get OIDC setup with ADFS running on Windows Server 2022. I have gotten as far as setting up the Application Group with the respective permitted scopes, but for some reason even though email is included in the permitted scopes I get the error on Outline about email not existing.

Outline Error Code:
```bash
{"error":"An email field was not returned in the profile parameter, but is required.","level":"error","message":"Error during authentication","stack":"UnauthorizedError: An email field was not returned in the profile parameter, but is required.\n    at AuthenticationError (/var/outline/build/server/errors.js:38:34)\n    at OAuth2Strategy._verify (/var/outline/build/plugins/oidc/server/auth/oidc.js:64:47)\n    at /var/outline/node_modules/passport-oauth2/lib/strategy.js:196:24\n    at _passportOauth.Strategy.userProfile (/var/outline/build/plugins/oidc/server/auth/oidc.js:24:12)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)"}
```

Outline Environment Configuration:
```
OIDC_CLIENT_ID=<redacted>
OIDC_CLIENT_SECRET=<redacted>
OIDC_AUTH_URI=https://<adfsServer>/adfs/oauth2/authorize
OIDC_TOKEN_URI=https://<adfsServer>/adfs/oauth2/token
OIDC_USERINFO_URI=https://<adfsServer>/adfs/userinfo
OIDC_USERNAME_CLAIM=email
OIDC_DISPLAY_NAME=<redacted>
OIDC_SCOPES="openid profile email"
```
<img width="403" alt="Screenshot 2023-08-31 at 11 53 02 AM" src="https://github.com/outline/outline/assets/34459904/0eb3a2be-99fe-45af-bd06-4349d5b435e1">

**Comment by tommoor** [comment]
Separate product, but I believe this step may be needed:

https://bitwarden.com/help/adfs-oidc-implementation/#add-a-transform-claim-rule

**Reply by PowellTravis** [reply]
I did the transform claim rule, but don't see how to apply the rest of the documentation to Outline. I still get the same error, the part 2 alterations don't work with Outlines OIDC implementation as from looking at the OIDC plugin code it only takes in the variables listed above.

**Reply by tommoor** [reply]
Yes I meant the transform claim rule. I’m not familiar with ADFS so can’t really help much further.

**Reply by PowellTravis** [reply]
No problem, thank you for the article.

**Comment by alricsans** [comment]
Hi @PowellTravis, I am running into the same issue, did you find a solution ? 

Thank you
